gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -281,6 +280,24 @@
             Optional.fromNullable(this.dataset.getProperties().getProperty(COPY_PARTITIONS_FILTER_CONSTANT));
       }
 
+      // Initialize extended partition filter
+      if ( this.dataset.getProperties().containsKey(PathBasedPartitionFilter.HIVE_PARTITION_EXTENDED_FILTER_TYPE)&&
```

Comment:
You could instead use a `ClassAliasResolver`. See for example `CliEmbeddedGobblin`. Instead of passing a String into the constructor, use a Factory that takes a `Config` as input. E.g.
```java
interface HivePartitionFilterFactory {
   HivePartitionFilter createFilter(Config config);
}
```

```java
interface HivePartitionFilter {
  boolean accept(Partition partition);
}
```

```java
new ClassAliasResolver<>(HivePartitionFilterFactory.class).resolveClass(filterName).newInstance().createFilter(config);
```